cpp/src/arrow/compute/kernels/aggregate_test.cc

```diff
@@ -2175,6 +2175,468 @@ TEST(TestFixedSizeBinaryMinMaxKernel, Basics) {
   EXPECT_THAT(MinMax(ScalarFromJSON(ty, R"("aa")"), options), ResultWith(null));
 }
 
+TEST(TestDictionaryMinMaxKernel, DecimalsValue) {
```

Comment:
IMO we have too many test cases, which will slow down compilation and CI. Could you reduce them to 2~3 test cases for each data type? We don't need to cover all test cases of their primitive kernels.

Reply:
I tend to preserve all the tests for one data type like decimal and reduce other data types. So we can still cover all cases in one type and have less burden on CI.